### PR TITLE
NMS-16510 : Search does not return any results for Asset Search strin…

### PR DESCRIPTION
**Pull request by Junaid-Khan-Nant**
- https://opennms.atlassian.net/browse/NMS-16510

### CODE REVIEW

**Review by cgorantla — COMMENTED** [review]
Please add a test case either in integration or smoke tests.

**Review by cgorantla — APPROVED** [review]
LGTM. Please also validate if all other asset related fields have right names in the Restrictions.

**Comment by Junaid-Khan-Nant** [comment]
Hi @cgorantla, All restrictions are checked; it seems fine except one that's maintContractExpires. But it is also working fine as we get value from getter; in that case, a screenshot is attached for reference. Don't need to change in it, I believe.
Screenshot from 2024-11-04 16-10-19

File: features/search/providers/pom.xml [hunk 1]
```diff
@@ -60,11 +60,30 @@
...
+    </dependency>
     <dependency>
       <groupId>org.opennms.core.ipc.rpc</groupId>
       <artifactId>org.opennms.core.ipc.rpc.mock-impl</artifactId>
       <version>${project.version}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.opennms.core.test-api</groupId>
+      <artifactId>org.opennms.core.test-api.db</artifactId>
```
Comment (cgorantla) [inline features/search/providers/pom.xml:86]: <scope>test</scope>

File: features/search/providers/src/test/java/org/opennms/netmgt/search/providers/node/NodeAssetSearchProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
+package org.opennms.netmgt.search.providers.node;
```
Comment (cgorantla) [inline features/search/providers/src/test/java/org/opennms/netmgt/search/providers/node/NodeAssetSearchProviderTest.java:1]: nit :missing copyright headers

**Review by cgorantla — APPROVED** [review]
LGTM!

**Pull request merged**